renku/core/dataset/providers/web.py

```diff
@@ -161,37 +157,15 @@ def download_files(
 
     destination.mkdir(parents=True, exist_ok=True)
 
-    listeners = communication.get_listeners()
-
-    def subscribe_communication_listeners(function, **kwargs):
-        try:
-            for communicator in listeners:
-                communication.subscribe(communicator)
-            return function(**kwargs)
-        finally:
-            for communicator in listeners:
-                communication.unsubscribe(communicator)
-
-    files = []
-    n_cpus = os.cpu_count() or 1
-    max_workers = min(n_cpus + 4, 8)
-    with concurrent.futures.ThreadPoolExecutor(max_workers) as executor:
-        futures = {
-            executor.submit(
-                subscribe_communication_listeners,
-                download_file,
-                project_path=project_context.path,
-                uri=url,
-                destination=destination,
-                extract=extract,
-                filename=name,
-                multiple=True,
-                delay=max_workers,  # NOTE: Rate limit to 1 request/second
-            )
-            for url, name in zip(urls, names)
-        }
-
-        for future in concurrent.futures.as_completed(futures):
-            files.extend(future.result())
-
-    return files
+    def download_file_wrapper(uri_filename: Tuple[str, Optional[str]], **kwargs):
```

Comment:
if you changed it do `*d` in `parallel_execute` this wrapper wouldn't be necessary, I think